### DISCUSSION TITLE
Why does Reticulum send an announcement upon starting?

### DISCUSSION THREAD

**Discussion by ThomasFreedman** [markqvist/Reticulum#423] [Help & Questions] [answered]
I have a simple announce monitor that prints incoming announces for my app in real time I use for debugging. RNS prints a callback error message from this monitor program when I start a separate program.

I've isolated the code in the separate program that triggers that error in the monitor, which is when the RNS IN destination is created in the programs constructor. If I exit the program before that I don't get the error. The error shows up tho when I exit right after this:

        RNS.Destination(rnsID, inOut, RNS.Destination.SINGLE, self.AppName, *aspects)

Why does RNS send an announce at this point?

The error is:
***[Error] The contained exception was: 'NoneType' object has no attribute 'decode'***

The variable app_data is 'None'. I have changed the monitor code to test for that now but I'm puzzled by why the announce is sent when the destination is created. I don't recall seeing this before now, but I suspect it has always been there to help RNS create a path for the destination.

IIRC I've questioned the need for announces to be made, and the answer I got was the node won't be discoverable otherwise. If an automatic announce is sent on destination initialization as appears to be the case, that doesn't seem necessary. Was RNS changed last year to send an announce automatically when a destination is created?

**Comment by markqvist** [comment, marked as answer]
> I have a simple announce monitor that prints incoming announces for my app in real time I use for debugging. RNS prints a callback error message from this monitor program when I start a separate program.

You have some sort of bug in the monitor code doing the printing. From the exception it sounds like something excepts a string (the "decode" method call), but gets a None (which is expected behaviour, and needs to be handled in your app).

> Why does RNS send an announce at this point?

RNS sends an announce from your app to the master RNS instance running on your system, so it will know what local application or shared instance client has your destination. The master instance has no way to snoop on other running processes, and the only way for it to know that you have actually spawned a new destination is if the respective app tells it about it.

This happens automatically when creating a destination on a system running a shared instance. The announce is local only, and not propagated to any external interfaces, unless the application in question actually calls `announce()` on the destination.

**Reply by markqvist** [reply]
To expand a little, the network connection between a master shared RNS instance and any client apps *is itself Reticulum*. It's just made transparent to the apps themselves with some extra internal logic in the Transport stack, but for all intents and purposes, it's Reticulum all the way down.

**Comment by ThomasFreedman** [comment]
You have some sort of bug in the monitor code doing the printing. 
             
Yes, as I mentioned that's fixed. Good to know such announce doesn't propagate outside of the machine. 

On another topic, did you ever resolve the issue you had with the Open Modems?, have you resumed shipping them?

**Reply by markqvist** [reply]
It seems stock levels for the required components are returning to normal, but currently I need to prioritise other things than making a production run of OpenModems. If there is enough interest, or an order of 50+ units, I can make them on demand at this point, though.
